borg/archiver.py

```diff
@@ -1399,7 +1440,7 @@ def build_parser(self, args=None, prog=None):
         subparser.add_argument('location', metavar='ARCHIVE',
                                type=location_validator(archive=True),
                                help='name of archive to create (must be also a valid directory name)')
-        subparser.add_argument('paths', metavar='PATH', nargs='+', type=str,
+        subparser.add_argument('paths', metavar='PATH', nargs='*', type=str,
```

Comment:
We could:

1. monkey-patch argparse: Unfortunately the bug resides in a closure inside a rather long method: https://hg.python.org/cpython/file/tip/Lib/argparse.py#l1771
We could replace the whole method or just the inner function (consume_positionals()): http://stackoverflow.com/a/27550228

2. use a hack to work around the bug, e.g. something like that:
```python
    import argparse

    class PatchedArgumentParser(argparse.ArgumentParser):
        def _parse_known_args(self, arg_strings, namespace):
            """work around a bug in argparse: https://bugs.python.org/issue15112"""
            arg = getattr(self, '_patch_arg', None)
            if arg:
                saved = arg.nargs, arg.required
                arg.nargs, arg.required = argparse.ONE_OR_MORE, False
            try:
                ret = super()._parse_known_args(arg_strings, namespace)
            finally:
                if arg:
                    arg.nargs, arg.required = saved
            return ret

    if __name__ == '__main__':
        parser = PatchedArgumentParser(prog='PROG')
        parser.add_argument('--foo', action='store_true')

        parser.add_argument('location')
        arg = parser.add_argument('paths', nargs='*')
        parser._patch_arg = arg

        parser.print_help()

        # args = parser.parse_args('1 --foo 2'.split())
        for argstr in ('1 --foo 2',
                       '--foo 1 2',
                       '1 --foo',
                       '--foo'):
            print(argstr)
            args = parser.parse_args(argstr.split())
            print("    {!r}".format(args))
```
3. do nothing and live with the fact that some commandlines have to be re-arranged